lib/chat/key_share.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Chat.KeyShare do
+  @moduledoc "Manipulate social sharing keys"
+
+  alias Chat.{Dialogs, Dialogs.Dialog, Identity}
+  alias Chat.{ChunkedFiles, ChunkedFilesMultisecret}
+  alias Chat.Upload.UploadKey
+
+  @threshold 4
+
+  def generate_key_shares({%Identity{private_key: private_key} = _me, users}) do
+    shares =
+      private_key
+      |> Enigma.hide_secret_in_shares(users |> Enum.count(), @threshold)
+
+    share_count = Enum.count(shares)
```

Comment:
User count, then shares count ... they should be the same

Reply:
change to use amount = Enum.count(users)